### PR TITLE
Update aie-public to llvmorg-19-init~1

### PR DESCRIPTION
**Pull request by konstantinschwarz**
This merges upstream commit 93248729cfae82a5ca2323d4a8e15aa3b9b9c707 into aie-public, resolves merge conflicts in the merge commit, and adds fixup commits on top to fix build and test errors.

### CODE REVIEW

File: llvm/test/CodeGen/AIE/GlobalISel/xfail-legalize-zextload.mir [hunk 1]
```diff
@@ -1,38 +0,0 @@
...
-#
-# (c) Copyright 2023-2024 Advanced Micro Devices, Inc. or its affiliates
-# RUN: not --crash llc -mtriple aie -run-pass=legalizer %s -verify-machineinstrs -o - 2>&1 | FileCheck %s
-
-# REQUIRES: asserts
-
-# The legalizer currently asserts when trying to lower a non-power of 2 memory sized zero extending load,
-# as its artifact combiner does not handle the G_ASSERT_ZEXT instruction introduced in the process.
-# This case isn't super interesting, but crashing isn't nice either.
-# TODO: properly implement lowering of non-power of 2 memory sized zero extending loads
```
Comment (gbossu) [inline llvm/test/CodeGen/AIE/GlobalISel/xfail-legalize-zextload.mir]: Yay!

File: llvm/test/CodeGen/AIE/aie2/sibling-constraint2.ll [hunk 2]
```diff
@@ -17,13 +17,13 @@
 define void @test(i8 %a, i32 %b) {
 entry:
   %cmp0 = icmp eq i32 %b, 0
-  %sel0 = select i1 %cmp0, i8 1, i8 0
+  %sel0 = select i1 %cmp0, i8 0, i8 2
```
Comment (gbossu) [inline llvm/test/CodeGen/AIE/aie2/sibling-constraint2.ll]: Why this change?

File: llvm/test/TableGen/GlobalISelEmitter-implicit-defs.td [hunk 3]
```diff
@@ -50,6 +50,7 @@ def : Pat<(sra (i32 GPR32:$rs1), (i32 GPR32:$rs2)),
 // CHECK-NEXT:       GIR_Copy, /*NewInsnID*/0, /*OldInsnID*/0, /*OpIdx*/0, // DstI[dst]
 // CHECK-NEXT:       GIR_Copy, /*NewInsnID*/0, /*OldInsnID*/0, /*OpIdx*/1, // rs1
 // CHECK-NEXT:       GIR_AddSimpleTempRegister, /*InsnID*/0, /*TempRegID*/0,
+// CHECK-NEXT:       GIR_ConstrainSelectedInstOperands, /*InsnID*/0,
```
Comment (gbossu) [inline llvm/test/TableGen/GlobalISelEmitter-implicit-defs.td]: I guess that new action can't hurt? I'm still struggling to remember how to read these files.

File: clang/test/CodeGen/aie/aie-abi-cpp.cpp [hunk 4]
```diff
@@ -1,4 +1,4 @@
-// NOTE: Assertions have been autogenerated by utils/update_cc_test_checks.py UTC_ARGS: --function-signature --return-type --skip-function-body
+// NOTE: Assertions have been autogenerated by utils/update_cc_test_checks.py UTC_ARGS: --function-signature
```
Comment (gbossu) [inline clang/test/CodeGen/aie/aie-abi-cpp.cpp]: Are `--return-type --skip-function-body` not needed anymore? Or did you manually update the test? If so, please remove the header like altogether.

**Review by gbossu — DISMISSED** [review]
The fixups look good to me. I think we should prioritize merging this before it gets out of hand.

File: llvm/test/CodeGen/AIE/aie2/sibling-constraint2.ll (hunk 2 above)
Comment (konstantinschwarz) [reply llvm/test/CodeGen/AIE/aie2/sibling-constraint2.ll]: GlobalISel has a new combiner for G_SELECT, which eliminates selects based on 0/1. Using different values doesn't trigger these combines

File: clang/test/CodeGen/aie/aie-abi-cpp.cpp (hunk 4 above)
Comment (konstantinschwarz) [reply clang/test/CodeGen/aie/aie-abi-cpp.cpp]: I had to update these tests manually, `--return-type --skip-function-body` doesn't exist in aie-public.
I'll drop these headers for now

**Review by stephenneuendorffer — APPROVED** [review]
